hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OmSnapshot.java

```diff
@@ -82,11 +91,30 @@ public OmSnapshot(KeyManager keyManager,
     this.bucketName = bucketName;
     this.volumeName = volumeName;
     this.omMetadataManager = keyManager.getMetadataManager();
+    this.snapshotDBKey = SnapshotInfo.getTableKey(
+        volumeName, bucketName, snapshotName);
+    this.activeOMMetadataManager = ozoneManager.getMetadataManager();
   }
 
+  /**
+   * Check snapshot status in active DB snapshotInfoTable.
+   * @throws IOException FileNotFoundException when snapshot is not active.
+   */
+  private void checkSnapshotStatus() throws IOException {
+    if (activeOMMetadataManager.getSnapshotInfoTable().get(snapshotDBKey)
```

Comment:
+1, However do we ever need to create any metadata manager object for a deleted but not yet reclaimed snapshot ?

Reply:
Right. I will do another take on this with @sadanand48 's suggestion. Block instance creation if snapshot is not ACTIVE, and close the instance on snapshot delete. Reference counting will be done in another jira to ensure DB instances in active use won't be closed.

Reply:
Hold on. I'll need to add an option to check whether this DB instance is going to be opened by a client or by SDT. The latter must be allowed.

Reply:
Addressed. With oen more TODO that needs to be addressed with HDDS-7935.